### ISSUE TITLE
DDC does not work with docking station that uses DisplayPort/MST on Mac

### ISSUE THREAD

**Issue by qkw24** [MonitorControl/MonitorControl#1403]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

I have a Dell S2421H connected via HP Thunderbolt Dock G2, the DDC control functionality does not work when using Monitor Control on Mac. However, when using the same dock and monitor on Windows 11, with apps like DisplayBuddy, Monitorian, the DDC control does work, and I did verify the display brightness can be adjusted using DDC. On first glance, it looks like the dock supports DDC, it's either macOS or Monitor Control preventing DDC from working.

### Steps to reproduce

1. Connect display to the display port output of the HP Thunderbolt Dock G2.
2. Install Monitor Control on Mac and complete the necessary set up steps. 
3. Try to adjust brightness or volume on Dell S2421H monitor using DDC.
4. Dell S2421H monitor does not respond to the adjustments made on Monitor Control.
5. Keeping the same monitor + dock setup, get a Windows 11 laptop and install apps like DisplayBuddy or Monitorian.
6. Observe that the brightness can be adjusted using DDC with Windows 11 + DisplayBuddy/Monitorian.

### Expected behavior

Monitor should respond to adjustments.

### Environment Information (please complete the following information)

```markdown
- macOS version: Ventura 13.4.1
- Mac model: MacBook Pro 14-inch, 2021, M1 Max
- MonitorControl version: 4.1.0
- Monitor(s): Dell S2421H
- Apple Silicon/M1 (yes or no): Yes
```

**Comment by Tommmciu** [comment]
Same story here with G2 dock, iiyama XUB2792QSU 
for DisplayPort DDC is not working, display connected via Thunderbolt->DisplayPort DDC is working. Where can I look for more information? any logs?

**Comment by waydabber** [comment]
Docks that use DisplayPort MST technology in any form won't properly work with Mac. DDC won't work and the secondary DisplayPort connector will just show the same screen as the primary one. The G2 dock is such unfortunately.

**Comment by qkw24** [comment]
Just to confirm, even we are not actively using DisplayPort MST (e.g. only having one monitor connected to the dock's DP port), DDC still won't work?

**Comment by waydabber** [comment]
Yes, this seems to be the case for all other devices as well.

**Comment by waydabber** [comment]
It should work except if the dock uses MST. MacOS generally does not support MST, probably there is some kind of internal addressing/routing issue (?) with i2c communication with hubs that use MST. Did not investigate the details though.

**Comment by sidevesh** [comment]
@waydabber I tested things MonitorControl with a displaylink dock and ddc does not work with that either,
on Windows things work normally for ddc control with a displaylink dock too.

**Comment by waydabber** [comment]
Yes, displaylink does not provide DDC API fro macOS. Did try to get in touch with them but it seems like they aren't too eager to get this fixed or provide an API for that.

**Comment by sidevesh** [comment]
@waydabber is there a thread or somewhere people can post and let displaylink folks know that this is something we need or can we show support in some other way,
and in the meantime keep this open, I think its a critical thing and sad that they aren't keen on implementing this when they have already done that for windows.

**Comment by sidevesh** [comment]
Also this issue also is about MST docks, which unless we are sure is impossible completely because of Mac OS just can't do this we should keep it open

**Comment by waydabber** [comment]
DisplayLink is a separate issue. I tried contacting them and offer to work with them on DDC but they just did not follow up, probably understaffed. Lack of MST support is a generic macOS issue. I'll keep this issue open as a "Known issue".

**Comment by Movsar-Khalakhoev** [comment]
any updates here?

**Comment by waydabber** [comment]
No.

**Comment by aktienautobahn** [comment]
I there any way to check if my dock station uses MST or not and if that is the reason only one monitor is being properly controlled using DDC? I am trying to understand what I have to change in my setup, so i can control multiple monitors using DDC and still having a one cable solution for host to dock over Thunderbolt.

My current setup is:
host: Macbook 14 M1 Pro Sonoma 14.5
dock: Belkin Thunderbolt 3 Dock Plus (two DP ports)
monitors: Samsung U28R55 4K

Currently state: Both monitor work properly, only settings like brightness control work only for one monitor on specific DP port on the dock. Also after plugging the second monitor directly to the host I can control both monitors using DDC.

Any ideas what I can do to control both monitors using DDC and still having one cable solution? (update dock station, cable)? 
Any ideas are appreciated!

**Comment by qkw24** [comment]
I think any dock that have more than one DP/HDMI ports supports MST.

Maybe you can try Thunderbolt 4 Docks with 2 Thunderbolt downstream ports in the back, and then you connect both monitors to the Thunderbolt ports, so that each monitor can hopefully be recognized as individual devices via the Thunderbolt channels.

**Comment by TeaRex-coder** [comment]
There's a [support page](https://support.displaylink.com/forums/287786-displaylink-feature-suggestions/suggestions/16135240-add-brightness-support-to-non-windows-oses) on the DisplayLink forum for DDC support on macOS which was started nearly 8 years ago!

In one of [MonitorControl's discussions](https://github.com/MonitorControl/MonitorControl/discussions/1580#discussioncomment-9978350), I shared a solution which might help get DDC to work with DisplayLink setups.
